src/vdaf.rs

```diff
@@ -272,43 +247,87 @@ pub struct VerifierMessage<F: FieldElement> {
+        /// indication matches the actual joint randomness seed.
+        //
+        // TODO(cjpatton) If `input.joint_rand_len() == 0`, then we don't need to bother with the
+        // joint randomness seed at all and make this optional.
+        joint_rand_seed: Key,
+    },
 }
 
-/// Run by each aggregator, this consumes the [`UploadMessage`] message sent from the client and
-/// produces the [`VerifierMessage`] message that will be broadcast to the other aggregators. It also
-/// returns the aggregator's state.
+/// The distributed state-initialization algorithm. This returns the initial state of each
+/// aggregator for evaluating the VDAF.
 ///
 /// # Parameters
 ///
-/// * `msg` is the message sent by the client. It contains, among other things, an input and proof
-/// share.
-/// * `param` is used to reconstruct the input share from the raw message.
-/// * `query_rand_seed` is used to derive the query randomness shared by all the aggregators.
-// TODO(cjpatton) Rename to dist_start to align with spec
-pub fn verify_start<V: Value>(
+/// * `suite` is the ciphersuite used for key derivation.
+/// * `input_param` is the parameter used to reconstruct a [`Value`] from a vector of field
+///   element.
+/// * `num_shares` is the number of aggregators.
+pub fn dist_init<V: Value>(
     suite: Suite,
-    msg: UploadMessage<V::Field>,
-    param: V::Param,
-    aggregator_id: u8,
-    query_rand_seed: &Key,
+    input_param: V::Param,
+    num_shares: u8,
+) -> Result<Vec<AggregatorState<V>>, VdafError> {
+    let query_rand_seed = Key::generate(suite)?;
+    Ok((0..num_shares)
+        .map(|aggregator_id| AggregatorState::Ready {
+            aggregator_id,
+            input_param: input_param.clone(),
+            query_rand_seed: query_rand_seed.clone(),
+        })
+        .collect())
+}
+
+/// The verify-start algorithm of the VDAF. Run by each aggregator, this consumes the
+/// [`InputShareMessage`] message sent from the client and produces the [`VerifierMessage`] message
+/// that will be broadcast to the other aggregators.
+// TODO(cjpatton) Check for ciphersuite mismatch between `state` and `msg`.
+pub fn verify_start<V: Value>(
+    state: AggregatorState<V>,
+    msg: InputShareMessage<V::Field>,
 ) -> Result<(AggregatorState<V>, VerifierMessage<V::Field>), VdafError> {
-    if query_rand_seed.suite() != suite {
-        return Err(VdafError::Uncategorized(
-            "verify_start(): joint rand seed type does not match suite".to_string(),
-        ));
-    }
+    let (aggregator_id, input_param, query_rand_seed) = match state {
+        AggregatorState::Ready {
+            aggregator_id,
+            input_param,
+            query_rand_seed,
+        } => (aggregator_id, input_param, query_rand_seed),
+        _ => {
+            return Err(VdafError::Uncategorized(
+                "verify_start(): called on incorrect state".to_string(),
+            ));
+        }
```

Comment:
nit: You could capture the invalid state in the error, if `AggregatorState` implemented `Debug`
```suggestion
        state => {
            return Err(VdafError::Uncategorized(
                format!("verify_start(): called on incorrect state {:?}", state),
            ));
        }
```

Reply:
Nice